src/full/Agda/Interaction/Options/Base.hs

```diff
@@ -409,62 +406,30 @@ unsafePragmaOptions clo opts =
   []
 
--- | If any these options have changed, then the file will be
---   rechecked. Boolean options are negated to mention non-default
---   options, where possible.
-
-restartOptions :: [(PragmaOptions -> RestartCodomain, String)]
-restartOptions =
-  [ (C . optTerminationDepth, "--termination-depth")
-  , (B . (/= UnicodeOk) . optUseUnicode, "--no-unicode")
-  , (B . optAllowUnsolved, "--allow-unsolved-metas")
-  , (B . optAllowIncompleteMatch, "--allow-incomplete-matches")
-  , (B . optDisablePositivity, "--no-positivity-check")
-  , (B . optTerminationCheck,  "--no-termination-check")
-  , (B . not . optUniverseCheck, "--type-in-type")
-  , (B . optOmegaInOmega, "--omega-in-omega")
-  , (B . optCumulativity, "--cumulativity")
-  , (B . collapseDefault . optSizedTypes, "--no-sized-types")
-  , (B . collapseDefault . optGuardedness, "--no-guardedness")
-  , (B . optInjectiveTypeConstructors, "--injective-type-constructors")
-  , (B . optProp, "--prop")
-  , (B . collapseDefault . optTwoLevel, "--two-level")
-  , (B . not . optUniversePolymorphism, "--no-universe-polymorphism")
-  , (B . optIrrelevantProjections, "--irrelevant-projections")
-  , (B . optExperimentalIrrelevance, "--experimental-irrelevance")
-  , (B . collapseDefault . optWithoutK, "--without-K")
-  , (B . collapseDefault . optCubicalCompatible, "--cubical-compatible")
-  , (B . optExactSplit, "--exact-split")
-  , (B . not . optEta, "--no-eta-equality")
-  , (B . optRewriting, "--rewriting")
-  , (B . (== Just CFull) . optCubical, "--cubical")
-  , (B . (== Just CErased) . optCubical, "--erased-cubical")
-  , (B . optGuarded, "--guarded")
-  , (B . optOverlappingInstances, "--overlapping-instances")
-  , (B . optQualifiedInstances, "--qualified-instances")
-  , (B . not . optQualifiedInstances, "--no-qualified-instances")
-  , (B . optSafe, "--safe")
-  , (B . optDoubleCheck, "--double-check")
-  , (M . optSyntacticEquality, "--syntactic-equality")
-  , (B . not . optAutoInline, "--no-auto-inline")
-  , (B . not . optFastReduce, "--no-fast-reduce")
-  , (B . optCallByName, "--call-by-name")
-  , (I . optInstanceSearchDepth, "--instance-search-depth")
-  , (I . optInversionMaxDepth, "--inversion-max-depth")
-  , (W . optWarningMode, "--warning")
-  , (B . (== Just LocalConfluenceCheck) . optConfluenceCheck, "--local-confluence-check")
-  , (B . (== Just GlobalConfluenceCheck) . optConfluenceCheck, "--confluence-check")
-  , (B . not . optImportSorts, "--no-import-sorts")
-  , (B . optAllowExec, "--allow-exec")
-  , (B . collapseDefault . optSaveMetas, "--save-metas")
-  , (B . optEraseRecordParameters, "--erase-record-parameters")
-  , (B . not . optLoadPrimitives, "--no-load-primitives")
-  ]
+-- | This function returns 'True' if the file should be rechecked.
 
--- to make all restart options have the same type
-data RestartCodomain
-  = C CutOff | B Bool | I Int | M !(Strict.Maybe Int) | W WarningMode
-  deriving Eq
+recheckBecausePragmaOptionsChanged
```

Comment:
Makes sense to whitelist rather than black-listing options (so the default is "recheck").
The documentation has a blacklist, so should that be updated?
https://github.com/agda/agda/blob/7588018496fbf489668bceb4ca99f401711053a2/doc/user-manual/tools/command-line-options.rst?plain=1#L1090-L1102
I also wondered how we ensure the doc is in sync with the code here...

Reply:
Does anyone rely on that list in the documentation? Perhaps it could be removed.

Reply:
Could be turned into a `--help` topic instead.

Reply:
I don't plan to do anything about this before the release. Feel free to open a new issue for your suggestions.